homeassistant/components/zyxelt50/device_tracker.py

```diff
@@ -0,0 +1,148 @@
+"""Support for Zyxel routers."""
+from __future__ import annotations
+
+from homeassistant.components.device_tracker import SOURCE_TYPE_ROUTER
+from homeassistant.components.device_tracker.config_entry import ScannerEntity
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+
+from .const import DATA_ZYXEL, DOMAIN
+from .router import Zyxel_T50_Router, ZyxelDevice
+
+
+async def async_setup_entry(
+        hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for Zyxel component."""
+    router = hass.data[DOMAIN][entry.entry_id][DATA_ZYXEL]
+    tracked = set()
+
+    @callback
+    def update_router():
+        """Update the values of the router."""
+        add_entities(router, async_add_entities, tracked)
+
+    router.async_on_close(
+        async_dispatcher_connect(hass, router.signal_device_new, update_router)
+    )
+
+    update_router()
+
+
+@callback
```

Comment:
This is not used as a callback, it is only called from a callback no need to mark it as a callback
```suggestion
```